django/cantusdb_project/main_app/views/chant.py

```diff
@@ -516,7 +515,7 @@ def get_queryset(self):
         search_text = self.request.GET.get("search_text")
 
         # get all chants in the specified source
-        chants = source.chant_set
+        chants = source.chant_set.select_related("feast", "office", "genre")
```

Comment:
I'm confused about what this is doing...where is this additional query bit used?

Reply:
These were necessary to include because many of the queries were caused in the chant_list.html template, specifically in:

```
                        </td>
                        <td class="text-wrap" style="text-align:center">
                            {% if chant.feast %}
                                <a href="{% url 'feast-detail' chant.feast.id %}" title="{{ chant.feast.description }}">{{ chant.feast.name|default:"" }}</a>
                            {% endif %}
                        </td>
                        <td class="text-wrap" style="text-align:center">
                            {% if chant.office %}
                                <a href="{% url 'office-detail' chant.office.id %}" title="{{ chant.office.description }}">{{ chant.office.name|default:"" }}</a>
                            {% endif %}
                        </td>
                        <td class="text-wrap" style="text-align:center">
                            {% if chant.genre %}
                                <a href="{% url 'genre-detail' chant.genre.id %}" title="{{ chant.genre.description }}">{{ chant.genre.name|default:"" }}</a>
                            {% endif %}
                        </td>
```

Reply:
Gotcha! Nice!